### PR TITLE
Allow using env var to manage snapshot creation and update

### LINKED ISSUE #125: Allow passing options as env var
Hi, I would like to use the package with paratest, which is used under the hood by Laravel for parallel testing. 

I don't succeed to pass any parameters like `--without-creating-snapshots`, and raised an issue: https://github.com/paratestphp/paratest/issues/607

The answer is:
>In Paratest the command is passed to PHPUnit directly, not through  var:
>
>https://github.com/paratestphp/paratest/blob/82d50dadbc357bc6a24c0a69710b6cae004017d5/bin/phpunit-wrapper.php#L41-L42
>
>I see you are leveraging a non orthodox way to get an option to the PHPUnit extension: the  should be reserved for  settings.
>
>Yes it would be possible to have  working on Paratest too, but I suggest you to ask/PR https://github.com/spatie/phpunit-snapshot-assertions to move away from  and rely on environment variables: they are widely used for this purpose (custom flags) and already supported both into  configuration and one-time runs.

And actually I'm quite agreeing with this statement: using env var for those parameters doesn't seem to be a bad idea (in addition of current way).

Would you be open to a PR for that?

Thanks.

### PR DESCRIPTION
**Pull request by mathieutu**
fix #125

### CODE REVIEW

**Comment by freekmurze** [comment]
Looks good! Could you also mention these new env vars in the readme?

**Comment by mathieutu** [comment]
I've also added a mention to parallel testing to highlight it, in case of some devs are looking for the solution of their problems about that.

File: src/MatchesSnapshots.php
Comment (freekmurze) [inline src/MatchesSnapshots.php:137]: For improved readability, we could break up this conditional and use early returns

**Review by freekmurze — CHANGES_REQUESTED** [review]

**Comment by mfn** [comment]
Random thought: would it make sense to promote using the env vars as being the _preferred_ way and adjust the code checks and language used in the README?

I figure it certainly is more compatible and maybe the thought experiment for deprecating the `-d --…` approach could be done?

**Comment by freekmurze** [comment]
> I figure it certainly is more compatible and maybe the thought experiment for deprecating the -d --… approach could be done?


I kinda like the `-d --` approach to quickly rewrite snapshots when a failing tests is fixed.

File: src/MatchesSnapshots.php
Comment (mathieutu) [inline src/MatchesSnapshots.php:138]: @freekmurze Actually no real early return possible here, except
```suggestion
		if(in_array('--update-snapshots', $_SERVER['argv'], true)) {
			return true
		}

        return getenv('UPDATE_SNAPSHOTS') === 'true';
```

Which is imo not really an improvement.

File: src/MatchesSnapshots.php
Comment (mathieutu) [reply src/MatchesSnapshots.php:137]: See my comments below about that.

File: src/MatchesSnapshots.php
Comment (freekmurze) [reply src/MatchesSnapshots.php:138]: To me this is more clear, thanks! I really like short conditionals

**Comment by freekmurze** [comment]
It seems like tests are failing, could you take a look at that? 

After the tests pass, I'll wait on a final stamp of approval by @sebastiandedeyne before merging and tagging this.

**Comment by mathieutu** [comment]
Actually suggestions are cools but no syntax checks or autocomplete! Un sem is missing 😅

**Comment by sebastiandedeyne** [comment]
5d9da943af014-stamp-of-approval

**Review by sebastiandedeyne — APPROVED** [review]

**Review by freekmurze — APPROVED** [review]

**Comment by freekmurze** [comment]
Thank you!

**Comment by mathieutu** [comment]
Always my pleasure!

**Pull request merged**
